armi/physics/fuelPerformance/tests/test_fuelPerformanceUtils.py

```diff
@@ -0,0 +1,36 @@
+"""
+Tests for fuel performance utilities.
+"""
+
+import unittest
+
+from armi.physics.fuelPerformance import utils
+from armi.reactor.flags import Flags
+from armi.reactor.tests import test_blocks
+
+
+class TestFuelPerformanceUtils(unittest.TestCase):
+    def test_enforceBondRemovalFraction(self):
+        b = test_blocks.loadTestBlock()
+        bond = b.getComponent(Flags.BOND)
+        bondRemovalFrac = 0.705
+        ndensBefore = b.getNumberDensity("NA")
+        bondNdensBefore = bond.getNumberDensity("NA")
+        b.p.bondBOL = bondNdensBefore
+        utils.enforceBondRemovalFraction(b, bondRemovalFrac)
+        bondNdensAfter = bond.getNumberDensity("NA")
+        ndensAfter = b.getNumberDensity("NA")
+
+        self.assertAlmostEqual(
+            bondNdensAfter / bondNdensBefore, (1.0 - bondRemovalFrac)
+        )
+        self.assertAlmostEqual(ndensBefore, ndensAfter)
+
+        # make sure it doesn't change if you run it twice
+        utils.enforceBondRemovalFraction(b, bondRemovalFrac)
+        bondNdensAfter = bond.getNumberDensity("NA")
+        ndensAfter = b.getNumberDensity("NA")
+        self.assertAlmostEqual(
+            bondNdensAfter / bondNdensBefore, (1.0 - bondRemovalFrac)
+        )
+        self.assertAlmostEqual(ndensBefore, ndensAfter)
```

Comment:
We don't currently have a system for validating _values_ of parameters, and I don't think it's appropriate to add it to this one-off parameter. Interesting feature request though.

Reply:
Sure, I think a general parameter validation could be a useful feature in the future. I didn't realize on the first pass that this test was copied from elsewhere and thought it was something new that was added.

Reply:
no it's just a move. I agree that that might be a useful feature, though I'm somewhat concerned about performance implications.

Reply:
Sure, I am concerned about the parameter definition saying that it must be between 0 and 1 inclusive and then there either being no test that demonstrates that the code fails outside of this range or that a more general framework validation functionality for testing ranges of parameters doesn't exist. It's especially concerning to me when a parameter is being set in an iterative/coupled fashion where it's either difficult to debug. Anyways, this can be made a future improvement to the testing and doesn't necessarily need to be fixed here.